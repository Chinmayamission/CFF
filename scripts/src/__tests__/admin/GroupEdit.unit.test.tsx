import { render } from "enzyme";
import React from "react";
import sinon from "sinon";
import GroupEdit from "../../admin/ResponseTable/GroupEdit";
import { responses, renderedForm } from "./constants";

it("renders class edit form", () => {
  const spy = sinon.spy();
  const dataOptionView = {
    id: "class_edit",
    displayName: "Edit Classes",
    groupEdit: "class"
  };
  const groupOption = {
    id: "class",
    schema: {
      type: "object",
      properties: {
        name: { type: "string" },
        room: { type: "string" },
        grade: { type: "string" },
        teacher: { type: "string" }
      }
    }
  };
  const wrapper = render(
    <GroupEdit
      groupOption={groupOption}
      dataOptionView={dataOptionView}
      onSubmit={spy}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain("Edit Classes");
  expect(wrapper.text()).toContain("name");
});
