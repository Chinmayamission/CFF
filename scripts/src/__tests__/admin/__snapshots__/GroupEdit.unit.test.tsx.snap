// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders class edit form 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    autocomplete="false"
    class="rjsf"
  >
    <div
      class="form-group field field-array"
    >
      <fieldset
        class="field field-array field-array-of-object"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Edit Classes
        </h2>
        <div
          class="array-item-list"
        >
          <div
            class="row"
          >
            <div
              class="col-12 ccmt-cff-array-item-container"
            >
              <div
                class="form-group field field-object"
              >
                <div
                  class="container-fluid"
                >
                  <span />
                  <div />
                  <div
                    class="row"
                  >
                    <div
                      class="form-group field field-string"
                    >
                      <label
                        class="control-label"
                        for="root_0_name"
                      >
                        name
                      </label>
                      <input
                        class="form-control"
                        id="root_0_name"
                        label="name"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="form-group field field-string"
                    >
                      <label
                        class="control-label"
                        for="root_0_room"
                      >
                        room
                      </label>
                      <input
                        class="form-control"
                        id="root_0_room"
                        label="room"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="form-group field field-string"
                    >
                      <label
                        class="control-label"
                        for="root_0_grade"
                      >
                        grade
                      </label>
                      <input
                        class="form-control"
                        id="root_0_grade"
                        label="grade"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="form-group field field-string"
                    >
                      <label
                        class="control-label"
                        for="root_0_teacher"
                      >
                        teacher
                      </label>
                      <input
                        class="form-control"
                        id="root_0_teacher"
                        label="teacher"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="btn btn-info col-12 mt-4 ccmt-cff-btn-array-add"
            type="button"
          >
            <small>
              <span
                class="oi oi-plus"
              />
            </small>
               Add
          </button>
        </div>
      </fieldset>
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;
